Phase I Deep CSV Training

mverzett opened the pull request:
This PR introduces the following:
   * DeepCMVA in the RECO Default content
   * DeepCSV/DeepCMVA in the miniAOD default content 
   * DeepCSV training for Phase I: removes the probcc discriminators and changes treatment of default values

This PR depends on [this](https://github.com/cms-data/RecoBTag-Combined/pull/5) in RecoBTag-Combined to work properly

Commit:
Phase I Deep CSV Training

cmsbuild commented:
A new Pull Request was created by  @mverzett (Mauro Verzetti) for master.

It involves the following packages:

PhysicsTools/PatAlgos
RecoBTag/Combined


@perrotta, @cmsbuild, @slava77, @monttj, @davidlange6 can you please review it and eventually sign? Thanks.
@TaiSakuma, @gouskos, @imarches, @ahinzmann, @acaudron, @mmarionncern, @rappoccio, @jdolen, @nhanvtran, @gpetruc, @gkasieczka, @schoef, @ferencek, @mverzett, @mariadalfonso, @pvmulder, @JyothsnaKomaragiri this is something you requested to watch as well.
@muzaffar, @davidlange6, @smuzaffar you are the release manager for this.

cms-bot commands are listed here cms-sw/cmssw#13028

slava77 commented:
@cmsbuild please test

cmsbuild commented:
The tests are being triggered in jenkins.

slava77 commented:
@cmsbuild abort tests

slava77 commented:
an external update in IB is needed (or at least a cmsdist PR with that update) before it can be tested

cmsbuild commented:
-1

Tested at: 2218370a5f335d9935932faa4338e11bd3c0577f

You can see the results of the tests here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-18315/19110/summary.html

I found follow errors while testing this PR

**Failed tests:  UnitTests  RelVals  AddOn**

* **Unit Tests**:

I found errors in the following unit tests: 
 
 ---> test runtestPhysicsToolsPatAlgos had ERRORS


* **RelVals**:

When I ran the RelVals I found an error in the following worklfows:
5.1 step1
<pre>runTheMatrix-results/5.1_TTbar+TTbarFS+HARVESTFS/step1_TTbar+TTbarFS+HARVESTFS.log

</pre>135.4 step1
<pre>runTheMatrix-results/135.4_ZEE_13+ZEEFS_13+HARVESTUP15FS+MINIAODMCUP15FS/step1_ZEE_13+ZEEFS_13+HARVESTUP15FS+MINIAODMCUP15FS.log

</pre>4.53 step3
<pre>runTheMatrix-results/4.53_RunPhoton2012B+RunPhoton2012B+HLTD+RECODR1reHLT+HARVESTDR1reHLT/step3_RunPhoton2012B+RunPhoton2012B+HLTD+RECODR1reHLT+HARVESTDR1reHLT.log

</pre>9.0 step3
<pre>runTheMatrix-results/9.0_Higgs200ChargedTaus+Higgs200ChargedTaus+DIGI+RECO+HARVEST/step3_Higgs200ChargedTaus+Higgs200ChargedTaus+DIGI+RECO+HARVEST.log

</pre>1306.0 step3
<pre>runTheMatrix-results/1306.0_SingleMuPt1_UP15+SingleMuPt1_UP15+DIGIUP15+RECOUP15+HARVESTUP15/step3_SingleMuPt1_UP15+SingleMuPt1_UP15+DIGIUP15+RECOUP15+HARVESTUP15.log

</pre>1000.0 step2
<pre>runTheMatrix-results/1000.0_RunMinBias2011A+RunMinBias2011A+TIER0+SKIMD+HARVESTDfst2+ALCASPLIT/step2_RunMinBias2011A+RunMinBias2011A+TIER0+SKIMD+HARVESTDfst2+ALCASPLIT.log

</pre>1330.0 step3
<pre>runTheMatrix-results/1330.0_ZMM_13+ZMM_13+DIGIUP15+RECOUP15+HARVESTUP15/step3_ZMM_13+ZMM_13+DIGIUP15+RECOUP15+HARVESTUP15.log

</pre>25.0 step3
<pre>runTheMatrix-results/25.0_TTbar+TTbar+DIGI+RECOAlCaCalo+HARVEST+ALCATT/step3_TTbar+TTbar+DIGI+RECOAlCaCalo+HARVEST+ALCATT.log

</pre>136.731 step3
<pre>runTheMatrix-results/136.731_RunSinglePh2016B+RunSinglePh2016B+HLTDR2_2016+RECODR2_2016reHLT_skimSinglePh_HIPM+HARVESTDR2/step3_RunSinglePh2016B+RunSinglePh2016B+HLTDR2_2016+RECODR2_2016reHLT_skimSinglePh_HIPM+HARVESTDR2.log

</pre>1001.0 step2
<pre>runTheMatrix-results/1001.0_RunMinBias2011A+RunMinBias2011A+TIER0EXP+ALCAEXP+ALCAHARVD1+ALCAHARVD2+ALCAHARVD3+ALCAHARVD4+ALCAHARVD5/step2_RunMinBias2011A+RunMinBias2011A+TIER0EXP+ALCAEXP+ALCAHARVD1+ALCAHARVD2+ALCAHARVD3+ALCAHARVD4+ALCAHARVD5.log

</pre>1003.0 step2
<pre>runTheMatrix-results/1003.0_RunMinBias2012A+RunMinBias2012A+RECODDQM+HARVESTDDQM/step2_RunMinBias2012A+RunMinBias2012A+RECODDQM+HARVESTDDQM.log

</pre>10042.0 step3
<pre>runTheMatrix-results/10042.0_ZMM_13+ZMM_13TeV_TuneCUETP8M1_2017_GenSimFull+DigiFull_2017+RecoFull_2017+ALCAFull_2017+HARVESTFull_2017/step3_ZMM_13+ZMM_13TeV_TuneCUETP8M1_2017_GenSimFull+DigiFull_2017+RecoFull_2017+ALCAFull_2017+HARVESTFull_2017.log

</pre>50202.0 step3
<pre>runTheMatrix-results/50202.0_TTbar_13+TTbar_13+DIGIUP15_PU50+RECOUP15_PU50+HARVESTUP15_PU50/step3_TTbar_13+TTbar_13+DIGIUP15_PU50+RECOUP15_PU50+HARVESTUP15_PU50.log

</pre>20034.0 step3
<pre>runTheMatrix-results/20034.0_TTbar_14TeV+TTbar_14TeV_TuneCUETP8M1_2023D7_GenSimHLBeamSpotFull14+DigiFullTrigger_2023D7+RecoFullGlobal_2023D7+HARVESTFullGlobal_2023D7/step3_TTbar_14TeV+TTbar_14TeV_TuneCUETP8M1_2023D7_GenSimHLBeamSpotFull14+DigiFullTrigger_2023D7+RecoFullGlobal_2023D7+HARVESTFullGlobal_2023D7.log

</pre>10024.0 step3
<pre>runTheMatrix-results/10024.0_TTbar_13+TTbar_13TeV_TuneCUETP8M1_2017_GenSimFull+DigiFull_2017+RecoFull_2017+ALCAFull_2017+HARVESTFull_2017/step3_TTbar_13+TTbar_13TeV_TuneCUETP8M1_2017_GenSimFull+DigiFull_2017+RecoFull_2017+ALCAFull_2017+HARVESTFull_2017.log

</pre>10824.0 step3
<pre>runTheMatrix-results/10824.0_TTbar_13+TTbar_13TeV_TuneCUETP8M1_2018_GenSimFull+DigiFull_2018+RecoFull_2018+ALCAFull_2018+HARVESTFull_2018/step3_TTbar_13+TTbar_13TeV_TuneCUETP8M1_2018_GenSimFull+DigiFull_2018+RecoFull_2018+ALCAFull_2018+HARVESTFull_2018.log

</pre>25202.0 step3
<pre>runTheMatrix-results/25202.0_TTbar_13+TTbar_13+DIGIUP15_PU25+RECOUP15_PU25+HARVESTUP15_PU25/step3_TTbar_13+TTbar_13+DIGIUP15_PU25+RECOUP15_PU25+HARVESTUP15_PU25.log

</pre>21234.0 step3
<pre>runTheMatrix-results/21234.0_TTbar_14TeV+TTbar_14TeV_TuneCUETP8M1_2023D4_GenSimHLBeamSpotFull14+DigiFullTrigger_2023D4+RecoFullGlobal_2023D4+HARVESTFullGlobal_2023D4/step3_TTbar_14TeV+TTbar_14TeV_TuneCUETP8M1_2023D4_GenSimHLBeamSpotFull14+DigiFullTrigger_2023D4+RecoFullGlobal_2023D4+HARVESTFullGlobal_2023D4.log

</pre>23234.0 step3
<pre>runTheMatrix-results/23234.0_TTbar_14TeV+TTbar_14TeV_TuneCUETP8M1_2023D8_GenSimHLBeamSpotFull14+DigiFullTrigger_2023D8+RecoFullGlobal_2023D8+HARVESTFullGlobal_2023D8/step3_TTbar_14TeV+TTbar_14TeV_TuneCUETP8M1_2023D8_GenSimHLBeamSpotFull14+DigiFullTrigger_2023D8+RecoFullGlobal_2023D8+HARVESTFullGlobal_2023D8.log

</pre>

* **AddOn**:

I found errors in the following addon tests:

cmsDriver.py TTbar_8TeV_TuneCUETP8M1_cfi  --conditions auto:run1_mc --fast  -n 100 --eventcontent AODSIM,DQM --relval 100000,1000 -s GEN,SIM,RECOBEFMIX,DIGI:pdigi_valid,L1,DIGI2RAW,L1Reco,RECO,EI,VALIDATION  --customise=HLTrigger/Configuration/CustomConfigs.L1THLT --datatier GEN-SIM-DIGI-RECO,DQMIO --beamspot Realistic8TeVCollision : FAILED - time: date Tue Apr 11 18:26:16 2017-date Tue Apr 11 18:24:50 2017 s - exit: 18688
cmsRun /build/cmsbld/jenkins-workarea/workspace/ib-any-integration/CMSSW_9_1_X_2017-04-11-1100/src/PhysicsTools/PatAlgos/test/IntegrationTest_cfg.py : FAILED - time: date Tue Apr 11 18:25:26 2017-date Tue Apr 11 18:24:53 2017 s - exit: 16640
cmsDriver.py RelVal -s HLT:Fake2,RAW2DIGI,L1Reco,RECO --data --scenario=pp -n 10 --conditions auto:run2_data_Fake2 --relval 9000,50 --datatier "RAW-HLT-RECO" --eventcontent FEVTDEBUGHLT --customise=HLTrigger/Configuration/CustomConfigs.L1THLT --era Run2_2016 --processName=HLTRECO --filein file:RelVal_Raw_Fake2_DATA.root --fileout file:RelVal_Raw_Fake2_DATA_HLT_RECO.root : FAILED - time: date Tue Apr 11 18:26:54 2017-date Tue Apr 11 18:24:55 2017 s - exit: 18688
cmsDriver.py RelVal -s HLT:PRef,RAW2DIGI,L1Reco,RECO --data --scenario=pp -n 10 --conditions auto:run2_data_PRef --relval 9000,50 --datatier "RAW-HLT-RECO" --eventcontent FEVTDEBUGHLT --customise=HLTrigger/Configuration/CustomConfigs.L1THLT --era Run2_2017 --magField 38T_PostLS1 --processName=HLTRECO --filein file:RelVal_Raw_PRef_DATA.root --fileout file:RelVal_Raw_PRef_DATA_HLT_RECO.root : FAILED - time: date Tue Apr 11 18:27:50 2017-date Tue Apr 11 18:24:59 2017 s - exit: 23040
cmsDriver.py RelVal -s HLT:GRun,RAW2DIGI,L1Reco,RECO --mc --scenario=pp -n 10 --conditions auto:run2_mc_GRun --relval 9000,50 --datatier "RAW-HLT-RECO" --eventcontent FEVTDEBUGHLT --customise=HLTrigger/Configuration/CustomConfigs.L1THLT --era Run2_2017 --magField 38T_PostLS1 --processName=HLTRECO --filein file:RelVal_Raw_GRun_MC.root --fileout file:RelVal_Raw_GRun_MC_HLT_RECO.root : FAILED - time: date Tue Apr 11 18:35:14 2017-date Tue Apr 11 18:25:02 2017 s - exit: 23040
cmsDriver.py TTbar_13TeV_TuneCUETP8M1_cfi --conditions auto:run2_mc --fast  -n 100 --eventcontent AODSIM,DQM --relval 100000,1000 -s GEN,SIM,RECOBEFMIX,DIGI:pdigi_valid,L1,DIGI2RAW,L1Reco,RECO,EI,VALIDATION  --customise=HLTrigger/Configuration/CustomConfigs.L1THLT --datatier GEN-SIM-DIGI-RECO,DQMIO --beamspot NominalCollision2015 --era Run2_25ns --magField 38T_PostLS1 : FAILED - time: date Tue Apr 11 18:26:24 2017-date Tue Apr 11 18:25:04 2017 s - exit: 18688
cmsDriver.py RelVal -s HLT:PRef,RAW2DIGI,L1Reco,RECO --mc --scenario=pp -n 10 --conditions auto:run2_mc_PRef --relval 9000,50 --datatier "RAW-HLT-RECO" --eventcontent FEVTDEBUGHLT --customise=HLTrigger/Configuration/CustomConfigs.L1THLT --era Run2_2017 --magField 38T_PostLS1 --processName=HLTRECO --filein file:RelVal_Raw_PRef_MC.root --fileout file:RelVal_Raw_PRef_MC_HLT_RECO.root : FAILED - time: date Tue Apr 11 18:33:19 2017-date Tue Apr 11 18:25:05 2017 s - exit: 23040
cmsDriver.py TTbar_13TeV_TuneCUETP8M1_cfi --conditions auto:run2_mc --fast  -n 100 --eventcontent AODSIM,DQM --relval 100000,1000 -s GEN,SIM,RECOBEFMIX,DIGI:pdigi_valid,L1,DIGI2RAW,L1Reco,RECO,EI,VALIDATION  --customise=HLTrigger/Configuration/CustomConfigs.L1THLT --datatier GEN-SIM-DIGI-RECO,DQMIO --beamspot NominalCollision2015 --era Run2_2016 --magField 38T_PostLS1 : FAILED - time: date Tue Apr 11 18:28:13 2017-date Tue Apr 11 18:25:36 2017 s - exit: 18688
cmsDriver.py RelVal -s HLT:GRun2016,RAW2DIGI,L1Reco,RECO --mc --scenario=pp -n 10 --conditions auto:run2_mc_GRun2016 --relval 9000,50 --datatier "RAW-HLT-RECO" --eventcontent FEVTDEBUGHLT --customise=HLTrigger/Configuration/CustomConfigs.L1THLT --era Run2_2016 --magField 38T_PostLS1 --processName=HLTRECO --filein file:RelVal_Raw_GRun2016_MC.root --fileout file:RelVal_Raw_GRun2016_MC_HLT_RECO.root : FAILED - time: date Tue Apr 11 18:36:06 2017-date Tue Apr 11 18:26:21 2017 s - exit: 18688
cmsDriver.py RelVal -s HLT:PIon,RAW2DIGI,L1Reco,RECO --data --scenario=pp -n 10 --conditions auto:run2_data_PIon --relval 9000,50 --datatier "RAW-HLT-RECO" --eventcontent FEVTDEBUGHLT --customise=HLTrigger/Configuration/CustomConfigs.L1THLT --era Run2_2017 --magField 38T_PostLS1 --processName=HLTRECO --filein file:RelVal_Raw_PIon_DATA.root --fileout file:RelVal_Raw_PIon_DATA_HLT_RECO.root : FAILED - time: date Tue Apr 11 18:31:04 2017-date Tue Apr 11 18:26:33 2017 s - exit: 23040
cmsDriver.py RelVal -s HLT:Fake,RAW2DIGI,L1Reco,RECO --mc --scenario=pp -n 10 --conditions auto:run1_mc_Fake --relval 9000,50 --datatier "RAW-HLT-RECO" --eventcontent FEVTDEBUGHLT --customise=HLTrigger/Configuration/CustomConfigs.L1THLT --processName=HLTRECO --filein file:RelVal_Raw_Fake_MC.root --fileout file:RelVal_Raw_Fake_MC_HLT_RECO.root : FAILED - time: date Tue Apr 11 18:34:18 2017-date Tue Apr 11 18:26:54 2017 s - exit: 18688
cmsDriver.py RelVal -s HLT:Fake2,RAW2DIGI,L1Reco,RECO --mc --scenario=pp -n 10 --conditions auto:run2_mc_Fake2 --relval 9000,50 --datatier "RAW-HLT-RECO" --eventcontent FEVTDEBUGHLT --customise=HLTrigger/Configuration/CustomConfigs.L1THLT --era Run2_2016 --processName=HLTRECO --filein file:RelVal_Raw_Fake2_MC.root --fileout file:RelVal_Raw_Fake2_MC_HLT_RECO.root : FAILED - time: date Tue Apr 11 18:35:46 2017-date Tue Apr 11 18:28:19 2017 s - exit: 18688
cmsDriver.py RelVal -s HLT:Fake1,RAW2DIGI,L1Reco,RECO --mc --scenario=pp -n 10 --conditions auto:run2_mc_Fake1 --relval 9000,50 --datatier "RAW-HLT-RECO" --eventcontent FEVTDEBUGHLT --customise=HLTrigger/Configuration/CustomConfigs.L1THLT --era Run2_25ns --processName=HLTRECO --filein file:RelVal_Raw_Fake1_MC.root --fileout file:RelVal_Raw_Fake1_MC_HLT_RECO.root : FAILED - time: date Tue Apr 11 18:37:30 2017-date Tue Apr 11 18:31:13 2017 s - exit: 18688
cmsDriver.py RelVal -s HLT:Fake,RAW2DIGI,L1Reco,RECO --data --scenario=pp -n 10 --conditions auto:run1_data_Fake --relval 9000,50 --datatier "RAW-HLT-RECO" --eventcontent FEVTDEBUGHLT --customise=HLTrigger/Configuration/CustomConfigs.L1THLT --processName=HLTRECO --filein file:RelVal_Raw_Fake_DATA.root --fileout file:RelVal_Raw_Fake_DATA_HLT_RECO.root : FAILED - time: date Tue Apr 11 18:35:57 2017-date Tue Apr 11 18:33:47 2017 s - exit: 18688
cmsDriver.py RelVal -s HLT:GRun,RAW2DIGI,L1Reco,RECO --data --scenario=pp -n 10 --conditions auto:run2_data_GRun --relval 9000,50 --datatier "RAW-HLT-RECO" --eventcontent FEVTDEBUGHLT --customise=HLTrigger/Configuration/CustomConfigs.L1THLT --era Run2_2017 --magField 38T_PostLS1 --processName=HLTRECO --filein file:RelVal_Raw_GRun_DATA.root --fileout file:RelVal_Raw_GRun_DATA_HLT_RECO.root : FAILED - time: date Tue Apr 11 18:38:57 2017-date Tue Apr 11 18:34:20 2017 s - exit: 23040
cmsDriver.py RelVal -s HLT:Fake1,RAW2DIGI,L1Reco,RECO --data --scenario=pp -n 10 --conditions auto:run2_data_Fake1 --relval 9000,50 --datatier "RAW-HLT-RECO" --eventcontent FEVTDEBUGHLT --customise=HLTrigger/Configuration/CustomConfigs.L1THLT --era Run2_25ns --processName=HLTRECO --filein file:RelVal_Raw_Fake1_DATA.root --fileout file:RelVal_Raw_Fake1_DATA_HLT_RECO.root : FAILED - time: date Tue Apr 11 18:36:26 2017-date Tue Apr 11 18:34:24 2017 s - exit: 18688
cmsDriver.py RelVal -s HLT:PIon,RAW2DIGI,L1Reco,RECO --mc --scenario=pp -n 10 --conditions auto:run2_mc_PIon --relval 9000,50 --datatier "RAW-HLT-RECO" --eventcontent FEVTDEBUGHLT --customise=HLTrigger/Configuration/CustomConfigs.L1THLT --era Run2_2017 --magField 38T_PostLS1 --processName=HLTRECO --filein file:RelVal_Raw_PIon_MC.root --fileout file:RelVal_Raw_PIon_MC_HLT_RECO.root : FAILED - time: date Tue Apr 11 18:40:11 2017-date Tue Apr 11 18:35:15 2017 s - exit: 23040
cmsDriver.py RelVal -s HLT:GRun2016,RAW2DIGI,L1Reco,RECO --data --scenario=pp -n 10 --conditions auto:run2_data_GRun2016 --relval 9000,50 --datatier "RAW-HLT-RECO" --eventcontent FEVTDEBUGHLT --customise=HLTrigger/Configuration/CustomConfigs.L1THLT --era Run2_2016 --magField 38T_PostLS1 --processName=HLTRECO --filein file:RelVal_Raw_GRun2016_DATA.root --fileout file:RelVal_Raw_GRun2016_DATA_HLT_RECO.root : FAILED - time: date Tue Apr 11 18:40:16 2017-date Tue Apr 11 18:35:56 2017 s - exit: 18688

cmsbuild commented:
Comparison not run due to runTheMatrix errors (RelVals and Igprof tests were also skipped)

slava77 commented:
looks like I misspelled the request to abort the tests and they eventually failed as expected due to the missing xml files

ferencek commented:
Given updates in `PhysicsTools/PatAlgos/python/producersLayer1/jetProducer_cfi.py`, `PhysicsTools/PatAlgos/python/patInputFiles_cff.py` will also have to be updated with input files that contain all the required discriminators. Also, please add the DeepFlavor taggers to `PhysicsTools/PatAlgos/test/patTuple_addBTagging_cfg.py` so that they are automatically included in PAT unit tests.

Commit:
apply Dinko's comments, bugfixes and naming convention fixes for the deepCSV/deepCMVA

cmsbuild commented:
Pull request #18315 was updated. @perrotta, @cmsbuild, @slava77, @monttj, @davidlange6 can you please check and sign again.

mverzett commented:
@ferencek I should have addressed your comments and a bit more that came in the way. For the test files I could not find a suitable AODSIM and therefore used a GEN-SIM-RECO

slava77 commented:
@cmsbuild please test with cms-sw/cmsdist#2954

cmsbuild commented:
The tests are being triggered in jenkins.
Using externals from cms-sw/cmsdist#2954

slava77 commented on PhysicsTools/PatAlgos/python/patInputFiles_cff.py:
```diff
@@ -24,7 +24,7 @@
                         #, useDAS        = True
                         #)
-        '/store/relval/CMSSW_8_0_0/RelValProdTTbar_13/AODSIM/80X_mcRun2_asymptotic_v4-v1/10000/DE81ABBF-1DDA-E511-8AF8-0026189438B5.root'
+        '/store/relval/CMSSW_9_1_0_pre2/RelValTTbar_13/GEN-SIM-RECO/90X_upgrade2017_realistic_v20-v2/00000/2257937F-3019-E711-BF48-0CC47A4D7678.root'
```
these better be AOD files
to match the purpose.

slava77 replied:
it seems more appropriate to define a new vstring for phase1 or otherwise also uniformly migrate PhysicsTools/PatAlgos/python/patTemplate_cfg.py to use 2017 GT and era

ferencek replied:
Would it make sense to have two sets of input files, one set for Phase 0 and another for Phase 1, set by an era modifier? For this to be of some use, we would need a version of `PhysicsTools/PatAlgos/python/patTemplate_cfg.py` that uses the 2017 GT and era. But wouldn't this imply duplicating all the PAT test cfg files? It would be nice to have some way of running all PAT tests in multiple eras but without duplicating all the cfg files.

davidlange6 replied:
If pat level workflows are now era dependent, then yes... [probably this breaks lots of analysis cfgs, no?..]

presumably that means have a few input arguments (era, file) that get passed down to configs..

slava77 commented on PhysicsTools/PatAlgos/test/patTuple_addBTagging_cfg.py:
```diff
@@ -126,18 +137,19 @@
    fatJets = cms.InputTag("ak8PFJetsCHS"),               # needed for subjet flavor clustering
-   groomedFatJets = cms.InputTag("ak8PFJetsCHSSoftDrop") # needed for subjet flavor clustering
+   groomedFatJets = cms.InputTag("ak8PFJetsCHSSoftDrop"), # needed for subjet flavor clustering
+   rParam = 0.8, # needed for subjet flavor clustering
```
was this reshuffling driven by some aesthetic reasons or was there a problem?

mverzett replied:
This was actually pretty useful for debugging an error. If you issue "step" to pdb on the previous configuration it actually enters the InputTag creation rather then the pat sequence modifier, hence I left it there in case we will need debugging of the sequence in future.

slava77 commented on RecoBTag/Combined/python/pfDeepCSVJetTags_cfi.py:
```diff
@@ -3,5 +3,12 @@
 	)
+
+from Configuration.Eras.Modifier_phase1Pixel_cff import phase1Pixel
+phase1Pixel.toModify(pfDeepCSVJetTags, NNConfig = cms.FileInPath('RecoBTag/Combined/data/DeepCSV_PhaseI.json'))
```
what is the plan to put these conditions in the GT?

ferencek replied:
Can json files such as this one used in the DeepCSV tagger be put in the CondDB? We can put GBRForest payloads in the CondDB for BDT-based taggers but for a general MVA technique, I'm not sure what are our options (of course, there are conditions used for the CSV algorithm based on a dedicated MVA computer code in `PhysicsTools/MVAComputer` but the general trend is to move away from that code since it is not very user-friendly and there are no longer developers around who really understand the inner workings of the code).

slava77 commented on RecoBTag/Combined/python/pfDeepCSVJetTags_cfi.py:
```diff
@@ -3,5 +3,12 @@
+from Configuration.Eras.Modifier_phase1Pixel_cff import phase1Pixel
+phase1Pixel.toModify(pfDeepCSVJetTags, NNConfig = cms.FileInPath('RecoBTag/Combined/data/DeepCSV_PhaseI.json'))
+phase1Pixel.toModify(pfDeepCSVJetTags, checkSVForDefaults = cms.bool(True))
+phase1Pixel.toModify(pfDeepCSVJetTags, meanPadding = cms.bool(True))
```
why are these parameters changing for 2017?
Other than 4 pixel layer related logic, the same best kind of training should be provided for 2017 and the past.

mverzett replied:
The model (line 12 and 14) has been explicitly trained for Phase I, and therefore should be used for Phase I only, there is no guarantee that the same training will perform better on Phase 0 as well.
Line 13 is a change in how the defaults are treated. We did not check the effect on the phase 0 training and we would like to keep it that way for compatibility reasons.

cmsbuild commented:
-1

Tested at: 8e18e0b9f7499f8a33759666eee492b038f4c452

You can see the results of the tests here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-18315/19157/summary.html

I found follow errors while testing this PR

**Failed tests:  UnitTests**

* **Unit Tests**:

I found errors in the following unit tests: 
 
 ---> test runtestPhysicsToolsPatAlgos had ERRORS

cmsbuild commented:
Comparison job queued.

ferencek commented on PhysicsTools/PatAlgos/test/patTuple_addBTagging_cfg.py:
```diff
@@ -101,6 +101,17 @@
     ,'pfChargeBJetTags'
+     #Deep Flavour
+    ,'pfDeepCSVJetTags:probb'
+    ,'pfDeepCSVJetTags:probc'
```
I guess you are aiming at the Phase I case only given the fact that there is no `probcc` here. This looks fine to me.

mverzett replied:
Well, it is more a "one size fit all" solution, this will work in both, and in the future the Phase 0 checks should be less and less relevant.

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-18315/19157/summary.html

The workflows 1003.0, 1001.0, 1000.0, 140.53, 136.731, 4.22 have different files in step1_dasquery.log than the ones found in the baseline. You may want to check and retrigger the tests if necessary. You can check it in the "files" directory in the results of the comparisons

Comparison Summary:
 - You potentially added 100 lines to the logs
 - ROOTFileChecks: Some differences in event products or their sizes found
 - Reco comparison results: 1657 differences found in the comparisons
 - DQMHistoTests: Total files compared: 23
 - DQMHistoTests: Total histograms compared: 1921491
 - DQMHistoTests: Total failures: 23869
 - DQMHistoTests: Total nulls: 0
 - DQMHistoTests: Total successes: 1897449
 - DQMHistoTests: Total skipped: 173
 - DQMHistoTests: Total Missing objects: 0
 - Checked 94 log files, 14 edm output root files, 23 DQM output files

slava77 commented:
On 4/13/17 10:48 AM, Mauro Verzetti wrote:
> The model (line 12 and 14) has been explicitly trained for Phase I, and
> therefore should be used for Phase I only, there is no guarantee that
> the same training will perform better on Phase 0 as well.
> Line 13 is a change in how the defaults are treated. We did not check
> the effect on the phase 0 training and we would like to keep it that way
> for compatibility reasons.

So, what are the specific observables that make this training method be
applicable only to Phase-1?
We should keep the methods the same.
It's fine to have the training files differ just because the training
inputs are for a different period,
but the approach and input variables better be the same.

It so far sounds like there was no time to apply this new method used
for phase-1 on phase-0/2016 samples;
hence the hacks added in the phase-1 modifier.

slava77 commented:
On 4/13/17 10:49 AM, Mauro Verzetti wrote:
> Well, it is more a "one size fit all" solution, this will work in both,
> and in the future the Phase 0 checks should be less and less relevant.
> 

the future includes support to processing of phase-0 data
(and even a rereco of 2016 at the end of run2)

imarches commented:
Dear all,

the Phase 1 DeepCSV training was actually tested by Markus on the 2016 geometry and only gives there a negligible improvement at the 1% level. We would not like to have the new Phase 1 training on the Legacy miniAODs. Our hope, once the Legacy files are there, is to cross check the Moriond SFs with two or three fast methods for the SF measurement and if the SFs are found to be identical to the Moriond SFs, as should likely be the case, we could simply recommend to use the Moriond SFs also for the Legacy files. This would avoid another campaign and we could focus the manpower on other tasks. Having a different version of DeepCSV may force to repeat all the analyses, if we want to provide SFs for that tagger with little uncertainties (it's quite some work, independently from the number of taggers that one commissions, as the selection etc has to be set up anyway). For a 1% gain it is not worth it, on our perspective.

slava77 commented:
On 4/13/17 10:56 AM, imarches wrote:
> Dear all,
> 
> the Phase 1 DeepCSV training was actually tested by Markus on the 2016
> geometry and only gives there a negligible improvement at the 1% level.
> We would not like to have the new Phase 1 training on the Legacy
> miniAODs. Our hope, once the Legacy files are there, is to cross check
> the Moriond SFs with two or three fast methods for the SF measurement
> and if the SFs are found to be identical to the Moriond SFs, as should
> likely be the case, we could simply recommend to use the Moriond SFs
> also for the Legacy files. This would avoid another campaign and we
> could focus the manpower on other tasks. Having a different version of
> DeepCSV may force to repeat all the analyses, if we want to provide SFs
> for that tagger with little uncertainties (it's quite some work,
> independently from the number of taggers that one commissions, as the
> selection etc has to be set up anyway). For a 1% gain it is not worth
> it, on our perspective.

It sounds like the comments above apply to 80X reco.
In this case, they are not relevant for this PR.

{Well, maybe I missed that there was some private reco of 2016 data with
91X for which the Moriond SFs mentioned above were derived.}


> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/cms-sw/cmssw/pull/18315#issuecomment-293975152>, or
> mute the thread
> <https://github.com/notifications/unsubscribe-auth/AEdcblZo13vOy2jSanF9txN-jFtBxkj6ks5rvmHMgaJpZM4M6Q5w>.
>

imarches commented:
Hi @slava77 to my knowledge there will be a Legacy 2016 production in 91X. When that happens we would still like not to modify the 2016 taggers. Hence, the switches in the code of this PR. The new Phase 1 developments in general do not apply to the old detector.

This we are doing consistently for all the taggers, charm tagger, double b tagger, as you have seen in the recent PRs.

slava77 commented:
On 4/13/17 11:32 AM, imarches wrote:
> Hi @slava77 <https://github.com/slava77> to my knowledge there will be a
> Legacy 2016 production in 91X. When that happens we would still like not
> to modify the 2016 taggers. Hence, the switches in the code of this PR.
> The new Phase 1 developments in general do not apply to the old detector.
> 
> This we are doing consistently for all the taggers, charm tagger, double
> b tagger, as you have seen in the recent PRs.

So, it sounds like you do not intend to support DNN taggers for 2016
MC/data in 91X
and will just keep the old ones to the extent that they still run
(despite changes in tracking, vertexing and other algorithms)
OK, I guess.


> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/cms-sw/cmssw/pull/18315#issuecomment-293984499>, or
> mute the thread
> <https://github.com/notifications/unsubscribe-auth/AEdcbsi49p3xzvuHwyNpUf0bWFODM-Llks5rvmohgaJpZM4M6Q5w>.
>

slava77 commented:
it looks like now phase-1 vs phase-0 is generating a difference in event content: probcc is present in phase-0 and not in phase-1.
I don't see a detector-driven reason to have an inconsistent treatment.
Please drop it uniformly everywhere (or maybe add some sub-optimal value in phase-1).

ferencek commented:
One consequence of the algorithmic changes introduced in Phase I is merging of the `probc` and `probcc` category into a single `probc` category. This is what causes differences in the event contents for Phase 0 and Phase I. I guess this is not allowed to happen in the same release cycle despite different eras.

@mverzett and the rest of the DeepFlavor team, would it be possible to simply sum up the Phase 0  `probc` and `probcc` values and store them in a single `probc` discriminator. As far as I understood, at least for now nobody really cares about having these two categories separately so by summing them up we would avoid introducing event content differences.

slava77 commented:
On 4/14/17 6:56 AM, Dinko Ferencek wrote:
> One consequence of the algorithmic changes introduced in Phase I is
> merging of the |probc| and |probcc| category into a single |probc|
> category. This is what causes differences in the event contents for
> Phase 0 and Phase I. I guess this is not allowed to happen in the same
> release cycle despite different eras.

No.
Can we just switch to the phase-1 training for everything?
It seems like the POG doesn't care about perfect correctness in 2016 and
earlier data/MC processing.
(and IIUC the phase-1 training behaves somewhat similarly on phase-0
inputs).

> 
> @mverzett <https://github.com/mverzett> and the rest of the DeepFlavor
> team, would it be possible to simply sum up the Phase 0 |probc| and
> |probcc| values and store them in a single |probc| discriminator. As far
> as I understood, at least for now nobody really cares about having these
> two categories separately so by summing them up we would avoid
> introducing event content differences.
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/cms-sw/cmssw/pull/18315#issuecomment-294157793>, or
> mute the thread
> <https://github.com/notifications/unsubscribe-auth/AEdcbk_I2xgz5wSzq6lESEq3LIfDowOrks5rv3sOgaJpZM4M6Q5w>.
>

mverzett commented:
@ferencek and @slava77 , would it be the same for you if in the Phase I training we added a ```probcc``` always set to zero? The effect would be the same but it would be easier to implement.

slava77 commented:
On 4/18/17 7:47 AM, Mauro Verzetti wrote:
> @ferencek <https://github.com/ferencek> and @slava77
> <https://github.com/slava77> , would it be the same for you if in the
> Phase I training we added a |probcc| always set to zero? The effect
> would be the same but it would be easier to implement.

I think that this is sub-optimal, but probably manageable for now.
I'd rather we eventually have a consistent content from physics point of
view for all the data periods.


> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/cms-sw/cmssw/pull/18315#issuecomment-294868329>, or
> mute the thread
> <https://github.com/notifications/unsubscribe-auth/AEdcbhzdo9U0rQ0JE5D1KkS20n1nZ01Gks5rxMzrgaJpZM4M6Q5w>.
>

Commit:
implement merging of the c and cc categories, make event content uniform

cmsbuild commented:
Pull request #18315 was updated. @perrotta, @cmsbuild, @slava77, @monttj, @davidlange6 can you please check and sign again.

slava77 commented:
@cmsbuild please test

cmsbuild commented:
The tests are being triggered in jenkins.

mverzett commented:
@slava77 @ferencek this last update should fix the event content making only ```probc``` available in the event content and summing the two probabilities for Phase 0

cmsbuild commented:
-1

Tested at: 7808091c804e443c82d8e2b884c4e1b0a83e1adf

You can see the results of the tests here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-18315/19247/summary.html

I found follow errors while testing this PR

**Failed tests:  UnitTests  RelVals  AddOn**

* **Unit Tests**:

I found errors in the following unit tests: 
 
 ---> test runtestPhysicsToolsPatAlgos had ERRORS


* **RelVals**:

When I ran the RelVals I found an error in the following worklfows:
10042.0 step3
<pre>runTheMatrix-results/10042.0_ZMM_13+ZMM_13TeV_TuneCUETP8M1_2017_GenSimFull+DigiFull_2017+RecoFull_2017+ALCAFull_2017+HARVESTFull_2017/step3_ZMM_13+ZMM_13TeV_TuneCUETP8M1_2017_GenSimFull+DigiFull_2017+RecoFull_2017+ALCAFull_2017+HARVESTFull_2017.log

</pre>10024.0 step3
<pre>runTheMatrix-results/10024.0_TTbar_13+TTbar_13TeV_TuneCUETP8M1_2017_GenSimFull+DigiFull_2017+RecoFull_2017+ALCAFull_2017+HARVESTFull_2017/step3_TTbar_13+TTbar_13TeV_TuneCUETP8M1_2017_GenSimFull+DigiFull_2017+RecoFull_2017+ALCAFull_2017+HARVESTFull_2017.log

</pre>20034.0 step3
<pre>runTheMatrix-results/20034.0_TTbar_14TeV+TTbar_14TeV_TuneCUETP8M1_2023D7_GenSimHLBeamSpotFull14+DigiFullTrigger_2023D7+RecoFullGlobal_2023D7+HARVESTFullGlobal_2023D7/step3_TTbar_14TeV+TTbar_14TeV_TuneCUETP8M1_2023D7_GenSimHLBeamSpotFull14+DigiFullTrigger_2023D7+RecoFullGlobal_2023D7+HARVESTFullGlobal_2023D7.log

</pre>10824.0 step3
<pre>runTheMatrix-results/10824.0_TTbar_13+TTbar_13TeV_TuneCUETP8M1_2018_GenSimFull+DigiFull_2018+RecoFull_2018+ALCAFull_2018+HARVESTFull_2018/step3_TTbar_13+TTbar_13TeV_TuneCUETP8M1_2018_GenSimFull+DigiFull_2018+RecoFull_2018+ALCAFull_2018+HARVESTFull_2018.log

</pre>21234.0 step3
<pre>runTheMatrix-results/21234.0_TTbar_14TeV+TTbar_14TeV_TuneCUETP8M1_2023D4_GenSimHLBeamSpotFull14+DigiFullTrigger_2023D4+RecoFullGlobal_2023D4+HARVESTFullGlobal_2023D4/step3_TTbar_14TeV+TTbar_14TeV_TuneCUETP8M1_2023D4_GenSimHLBeamSpotFull14+DigiFullTrigger_2023D4+RecoFullGlobal_2023D4+HARVESTFullGlobal_2023D4.log

</pre>23234.0 step3
<pre>runTheMatrix-results/23234.0_TTbar_14TeV+TTbar_14TeV_TuneCUETP8M1_2023D8_GenSimHLBeamSpotFull14+DigiFullTrigger_2023D8+RecoFullGlobal_2023D8+HARVESTFullGlobal_2023D8/step3_TTbar_14TeV+TTbar_14TeV_TuneCUETP8M1_2023D8_GenSimHLBeamSpotFull14+DigiFullTrigger_2023D8+RecoFullGlobal_2023D8+HARVESTFullGlobal_2023D8.log

</pre>

* **AddOn**:

I found errors in the following addon tests:

cmsDriver.py RelVal -s HLT:PRef,RAW2DIGI,L1Reco,RECO --data --scenario=pp -n 10 --conditions auto:run2_data_PRef --relval 9000,50 --datatier "RAW-HLT-RECO" --eventcontent FEVTDEBUGHLT --customise=HLTrigger/Configuration/CustomConfigs.L1THLT --era Run2_2017 --magField 38T_PostLS1 --processName=HLTRECO --filein file:RelVal_Raw_PRef_DATA.root --fileout file:RelVal_Raw_PRef_DATA_HLT_RECO.root : FAILED - time: date Wed Apr 19 15:01:58 2017-date Wed Apr 19 14:56:59 2017 s - exit: 23040
cmsDriver.py RelVal -s HLT:GRun,RAW2DIGI,L1Reco,RECO --mc --scenario=pp -n 10 --conditions auto:run2_mc_GRun --relval 9000,50 --datatier "RAW-HLT-RECO" --eventcontent FEVTDEBUGHLT --customise=HLTrigger/Configuration/CustomConfigs.L1THLT --era Run2_2017 --magField 38T_PostLS1 --processName=HLTRECO --filein file:RelVal_Raw_GRun_MC.root --fileout file:RelVal_Raw_GRun_MC_HLT_RECO.root : FAILED - time: date Wed Apr 19 15:07:02 2017-date Wed Apr 19 14:57:07 2017 s - exit: 23040
cmsDriver.py RelVal -s HLT:PRef,RAW2DIGI,L1Reco,RECO --mc --scenario=pp -n 10 --conditions auto:run2_mc_PRef --relval 9000,50 --datatier "RAW-HLT-RECO" --eventcontent FEVTDEBUGHLT --customise=HLTrigger/Configuration/CustomConfigs.L1THLT --era Run2_2017 --magField 38T_PostLS1 --processName=HLTRECO --filein file:RelVal_Raw_PRef_MC.root --fileout file:RelVal_Raw_PRef_MC_HLT_RECO.root : FAILED - time: date Wed Apr 19 15:07:00 2017-date Wed Apr 19 14:57:14 2017 s - exit: 23040
cmsDriver.py RelVal -s HLT:PIon,RAW2DIGI,L1Reco,RECO --data --scenario=pp -n 10 --conditions auto:run2_data_PIon --relval 9000,50 --datatier "RAW-HLT-RECO" --eventcontent FEVTDEBUGHLT --customise=HLTrigger/Configuration/CustomConfigs.L1THLT --era Run2_2017 --magField 38T_PostLS1 --processName=HLTRECO --filein file:RelVal_Raw_PIon_DATA.root --fileout file:RelVal_Raw_PIon_DATA_HLT_RECO.root : FAILED - time: date Wed Apr 19 15:07:22 2017-date Wed Apr 19 15:02:04 2017 s - exit: 23040
cmsDriver.py RelVal -s HLT:GRun,RAW2DIGI,L1Reco,RECO --data --scenario=pp -n 10 --conditions auto:run2_data_GRun --relval 9000,50 --datatier "RAW-HLT-RECO" --eventcontent FEVTDEBUGHLT --customise=HLTrigger/Configuration/CustomConfigs.L1THLT --era Run2_2017 --magField 38T_PostLS1 --processName=HLTRECO --filein file:RelVal_Raw_GRun_DATA.root --fileout file:RelVal_Raw_GRun_DATA_HLT_RECO.root : FAILED - time: date Wed Apr 19 15:12:11 2017-date Wed Apr 19 15:07:21 2017 s - exit: 23040
cmsDriver.py RelVal -s HLT:PIon,RAW2DIGI,L1Reco,RECO --mc --scenario=pp -n 10 --conditions auto:run2_mc_PIon --relval 9000,50 --datatier "RAW-HLT-RECO" --eventcontent FEVTDEBUGHLT --customise=HLTrigger/Configuration/CustomConfigs.L1THLT --era Run2_2017 --magField 38T_PostLS1 --processName=HLTRECO --filein file:RelVal_Raw_PIon_MC.root --fileout file:RelVal_Raw_PIon_MC_HLT_RECO.root : FAILED - time: date Wed Apr 19 15:16:37 2017-date Wed Apr 19 15:11:40 2017 s - exit: 23040

cmsbuild commented:
Comparison not run due to runTheMatrix errors (RelVals and Igprof tests were also skipped)

mverzett commented:
Unit tests are broken because I forgot to update all the test files (fix incoming). The matrix errors seem to be due to the lack of the [training json file](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-18315/19247/runTheMatrix-results/20034.0_TTbar_14TeV+TTbar_14TeV_TuneCUETP8M1_2023D7_GenSimHLBeamSpotFull14+DigiFullTrigger_2023D7+RecoFullGlobal_2023D7+HARVESTFullGlobal_2023D7/step3_TTbar_14TeV+TTbar_14TeV_TuneCUETP8M1_2023D7_GenSimHLBeamSpotFull14+DigiFullTrigger_2023D7+RecoFullGlobal_2023D7+HARVESTFullGlobal_2023D7.log). Has it been deployed?
An additional quirk is that from the [global log file](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-18315/19247/runTheMatrix-results/runall-report-step123-.log) it seems that the crash is happening on the step2, while the actual exception is in the step3 log and the step2 seems to work fine. Might be worth looking into it.

slava77 commented:
@mverzett 
I should have requested ``` @cmsbuild please test with cms-sw/cmsdist#2954 ```
this explains the matrix test problems

Commit:
update more pat tests input files

cmsbuild commented:
Pull request #18315 was updated. @perrotta, @cmsbuild, @slava77, @monttj, @davidlange6 can you please check and sign again.

slava77 commented:
@cmsbuild please test with cms-sw/cmsdist#2954

the cmsdist PR is merged already, but it will not appear in the used IB until after -2300 IB shows up today.

cmsbuild commented:
The tests are being triggered in jenkins.
Using externals from cms-sw/cmsdist#2954

cmsbuild commented:
-1

Tested at: 39fb835d31b3b4ece9ce9422a1dc049e22dacc9c

You can see the results of the tests here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-18315/19263/summary.html

I found follow errors while testing this PR

**Failed tests:  UnitTests**

* **Unit Tests**:

I found errors in the following unit tests: 
 
 ---> test runtestPhysicsToolsPatAlgos had ERRORS

cmsbuild commented:
Comparison job queued.

slava77 commented:
the unit test patTuple_updateJets_fromMiniAOD_cfg.py
is failing in a  place related to changes in this PR

```
   [6] Calling method for module JetCorrFactorsProducer/'patJetCorrFactorsSoftDropSubjets'
Exception Message:
Principal::getByToken: Found zero products matching all criteria
Looking for a container with elements of type: reco::Jet
Looking for module label: slimmedJetsAK8PFCHSSoftDropPacked
Looking for productInstanceName: SubJets

```

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-18315/19263/summary.html

Comparison Summary:
 - You potentially added 55 lines to the logs
 - ROOTFileChecks: Some differences in event products or their sizes found
 - Reco comparison results: 1698 differences found in the comparisons
 - DQMHistoTests: Total files compared: 23
 - DQMHistoTests: Total histograms compared: 1826239
 - DQMHistoTests: Total failures: 20557
 - DQMHistoTests: Total nulls: 0
 - DQMHistoTests: Total successes: 1805509
 - DQMHistoTests: Total skipped: 173
 - DQMHistoTests: Total Missing objects: 0
 - Checked 94 log files, 14 edm output root files, 23 DQM output files

Commit:
change the choice of test files and input collections for AK8 subjets

cmsbuild commented:
Pull request #18315 was updated. @perrotta, @cmsbuild, @slava77, @monttj, @davidlange6 can you please check and sign again.

slava77 commented:
@cmsbuild please test

cmsbuild commented:
The tests are being triggered in jenkins.

cmsbuild commented:
+1
Tested at: c9793cbc6fc15b6f4dad75b1de278426fb1c6ba8
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-18315/19296/summary.html

cmsbuild commented:
Comparison job queued.

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-18315/19296/summary.html

Comparison Summary:
 - You potentially added 34 lines to the logs
 - ROOTFileChecks: Some differences in event products or their sizes found
 - Reco comparison results: 1765 differences found in the comparisons
 - DQMHistoTests: Total files compared: 23
 - DQMHistoTests: Total histograms compared: 1826273
 - DQMHistoTests: Total failures: 35987
 - DQMHistoTests: Total nulls: 0
 - DQMHistoTests: Total successes: 1790113
 - DQMHistoTests: Total skipped: 173
 - DQMHistoTests: Total Missing objects: 0
 - Checked 94 log files, 14 edm output root files, 23 DQM output files

slava77 commented:
+1

for https://github.com/cms-sw/cmssw/pull/18315	c9793cb
- implementation is in line with the description; a follow up is suggested in #18443 to possibly apply the new training logic in phase-0 as well
- jenkins tests pass and comparisons with baseline show differences in expected places in DeepCMVA tagger plots
    - looking at a copy of jenkins outputs, there are 5 new b-tagging products in pfDeepCMVAJetTags_prob\* branches and some minor increase in slimmed\*jets in miniAOD corresponding to the addition of deepCSV tags
    - plots of pfDeepCMVAJetTags_prob\* look sensible based on simple inclusive distributions
